app/src/main/java/com/example/cinemapp/util/mappers/ActorDetailsMapper.kt

```diff
@@ -0,0 +1,61 @@
+package com.example.cinemapp.util.mappers
+
+import com.example.cinemapp.data.CastMovieCreditDTO
+import com.example.cinemapp.data.PersonDetailsDTO
+import com.example.cinemapp.ui.main.model.CastMovieCredit
+import com.example.cinemapp.ui.main.model.PersonDetails
+import java.time.LocalDate
+
+class ActorDetailsMapper(
+    private val urlMapper: URLMapper
+) {
+    fun mapDateStringListToDescendingYearsList(dates: List<String?>?): List<Int?> {
+        return dates?.map { date ->
+            if (date.isNullOrEmpty()) null
+            else date.substring(0, 4).toInt()
+        }?.distinct()?.sortedByDescending { year -> year ?: 0 } ?: emptyList()
+    }
+
+    private fun mapGenderIdToString(gender: Int): String {
```

Comment:
Having a when expression that maps one data type into a particular set of values tells us that we could use an `enum class` here. But lets just have this in mind and do a small refactor later 👍

Reply:
The problem with enum here was that the value for gender that you get from the network call is an int. When i put the type as an enum, the value doesn't initialize at all.

Reply:
Sure, we'll just skip this for now :D you can resolve this